pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/TableRebalancer.java

```diff
@@ -257,47 +390,162 @@ private void setTargetState(IdealState idealState, String segmentId, Map<String,
   }
 
   /**
-   * Check if IdealState = ExternalView. If its not equal, return the number of differing segments.
+   * Check if external view has converged to ideal state
+   * @param tableName name of table that we are rebalancing
+   * @return true if external view is same as ideal state, false otherwise
    */
-  public int isStable(String tableName) {
+  private boolean isStable(String tableName) {
```

Comment:
If we return a count of number of segments converged, that can show a progress bar. I am fine if that is a subsequent PR

Reply:
Sounds good.

Reply:
Done in the same PR -- keeping track of the number of segments not converged and then printing a log message